src/components/AddPresetAsset.tsx

```diff
@@ -0,0 +1,180 @@
+import { useHistory } from "react-router-dom";
+import {
+  Button,
+  Checkbox,
+  Heading2,
+  InfoBlock,
+  InfoBlockVariant,
+  Loader,
+} from "@stellar/design-system";
+import { TextLink } from "components/TextLink";
+import { presetAsset } from "constants/presetAssets";
+import { getErrorMessage } from "helpers/getErrorMessage";
+import { getNetworkConfig } from "helpers/getNetworkConfig";
+import { getPresetAssets } from "helpers/getPresetAssets";
+import { getValidatedUntrustedAsset } from "helpers/getValidatedUntrustedAsset";
+import { log } from "helpers/log";
+import { searchParam } from "helpers/searchParam";
+import { useRedux } from "hooks/useRedux";
+import { ActionStatus, SearchParams } from "types/types.d";
+import { shortenStellarKey } from "helpers/shortenStellarKey";
+
+export const AddPresetAsset = ({ onClose }: { onClose: () => void }) => {
+  const { account, allAssets, settings, untrustedAssets } = useRedux(
+    "account",
+    "allAssets",
+    "settings",
+    "untrustedAssets",
+  );
+  const [presetAssets, setPresetAssets] = useState<presetAsset[]>([]);
+  const [checkedAssets, setCheckedAssets] = useState<{
+    [key: string]: boolean;
+  }>({});
+  const [isValidating, setIsValidating] = useState(false);
+  const [errorMessage, setErrorMessage] = useState("");
+
+  const history = useHistory();
+
+  useEffect(() => {
+    setPresetAssets(getPresetAssets(allAssets.data));
+  }, [allAssets]);
+
+  useEffect(() => {
+    if (untrustedAssets.status === ActionStatus.SUCCESS) {
+      onClose();
+    }
+
+    if (untrustedAssets.errorString) {
+      setErrorMessage(untrustedAssets.errorString);
+    }
+  }, [onClose, untrustedAssets.errorString, untrustedAssets.status]);
+
+  const getAssetId = (asset: presetAsset) =>
+    `${asset.assetCode}:${asset.homeDomain || asset.issuerPublicKey}`;
+
+  const handleOnChangeCheckbox = (asset: presetAsset) => {
+    const updatedCheckedAssets = JSON.parse(JSON.stringify(checkedAssets));
+    const assetId = getAssetId(asset);
+    updatedCheckedAssets[assetId] = !checkedAssets[assetId];
+    setCheckedAssets(updatedCheckedAssets);
+  };
+
+  const handleAddUntrustedAssets = async (assetList: presetAsset[]) => {
+    setErrorMessage("");
+    setIsValidating(true);
+
+    try {
+      const validatedAssetValues = await Promise.all(
+        assetList.map(async (pAsset) => {
+          const { assetCode, issuerPublicKey, homeDomain } = pAsset;
+
+          if (!(homeDomain || issuerPublicKey)) {
+            const errorMsg = `Home domain OR issuer public key is required with asset code ${assetCode}`;
+            log.error({ title: errorMsg });
+            throw new Error(errorMsg);
+          }
+
+          const asset = await getValidatedUntrustedAsset({
+            assetCode,
+            homeDomain,
+            issuerPublicKey,
+            accountBalances: account.data?.balances,
+            networkUrl: getNetworkConfig(settings.pubnet).url,
```

Comment:
Do we need to use pubnet here?

Reply:
Yup! This `pubnet` is a boolean, it's set to true for pubnet and false for testnet:
https://github.com/stellar/stellar-demo-wallet/blob/790e8ba3812ee9bb5ebf73ffbce19f8ea2e674b8/src/types/types.d.ts#L130

Reply:
Can we want to only allow this for testnet? I don't think it makes sense to run this on mainnet.

Reply:
My bad, @marcelosalloum , I read it wrong after being away from this project for a while. Maybe we should use a boolean type of name for this (`isPubnet`). 🤔 Not related to this PR, of course.

Reply:
Good point, Shannon! With [this commit](https://github.com/stellar/stellar-demo-wallet/pull/206/commits/65ee3f6ee9589848ab0773d4349b01dbf468d325), we now hide that button if in pubnet.